### ISSUE TITLE
Clarification on how to provide AWS credentials for AWS S3 buckets

### ISSUE THREAD

**Issue by marcodelapierre** [nextflow-io/nextflow#4555]
Hi @pditommaso ,

I was testing NF+Wave+Fusion on a Ubuntu VM with my toy pipeline https://github.com/marcodelapierre/toy-acacia-nf .

In order for the S3 buckets to be properly used, it was not enough to setup the AWS `credentials`/`config` under `~/.aws`, and to add the config `fusion.exportStorageCredentials = true`.  
In fact, I also had to provide the credentials via the `nextflow.config` using the `aws` scope, for instance:
```
aws {
    accessKey = "$AWS_ACCESS_KEY_ID"
    secretKey = "$AWS_SECRET_ACCESS_KEY"
  }
```
(see https://github.com/marcodelapierre/toy-acacia-nf/blob/main/nextflow.config#L42-L43)

When I omitted them, the pipeline just hung for minutes with no errors, before I killed it.

Is this the correct behaviour?

**Comment by pditommaso** [comment]
Yes it's expected. Above all, it's a temporary solution, mostly for testing. A better mechanism not leaking the creds in the environment should be found

**Comment by marcodelapierre** [comment]
> Above all, it's a temporary solution, mostly for testing. A better mechanism not leaking the creds in the environment should be found.

How is it done for AWS Batch? Cannot it be setup the same way?

**Comment by marcodelapierre** [comment]
Also, I notice two inconsistencies with the Nextflow/Fusion documentation:

1. S3 Storage page (https://www.nextflow.io/docs/latest/amazons3.html#aws-access-and-secret-keys) lists 6 mechanisms to provide AWS credentials for S3; I have tested all of them except for IAM (no 6), and only no. 1 seems to work (nextflow.config), whereas AWS variables (2 and 3) and .aws config files (4 and 5) do not (various types of issues/errors). This is in the case of S3 + Fusion, both with and without AWS Batch.

2. Fusion page for the Local executor (https://www.nextflow.io/docs/latest/fusion.html#local-execution-with-s3-bucket-as-work-directory) suggests to use AWS env vars for access; however this does not work, and nextflow.config must be used instead.

**Key question**: In the cases above, is it possible to fix the code? Otherwise we should fix the docs.

(note: I did not test a case of S3 without Fusion on AWS, as I had no container with aws cli ready to go - could do in the future)
